Add scripts to run LOCO fitting & set fitted-strengths to the machine

vellosok75 opened the pull request:
Currently, LOCO-fitted strengths are applied to the machine via the `as_magnets_manipulate_strengths.ipynb` notebook. However, this notebook includes also other functionalities and a particular workflow which increases the risk of operational mistakes. This PR introduces a dedicated script to streamline the process, reduce human error, standardize and simplify the setting up of strengths.

Partner of https://github.com/lnls-fac/apsuite/pull/297, https://github.com/lnls-fac/apsuite/pull/299

Also includes script to run LOCO with proper input parsing. This script has been used for years but not tracked in any repository. See the helper:

![image](https://github.com/user-attachments/assets/09ac1a0d-d0e4-4f1b-9b01-833fd51c5f7e)

anacso17 commented:
@vellosok75 , please, update version to 0.21.0

fernandohds564 commented on bin/sirius-script-si-set-loco-strengths.py:
```diff
@@ -0,0 +1,224 @@
+        sys.exit(0)
+    apply_func()
+    if ask_yes_no("Check if applied?"):
+        print("Reapplying...")
```
why `"Reapplying"`? isn't this the first time the strengthss will be applied?

vellosok75 replied:
The changes are first applied by `apply_func`, in line 80. These functions have this `print_change` arg which, when true, prints out the strengths that will be applied to the magnets in order to reach the desired (fitted) strengths. If zeros are printed, we know it's all been set.  In other words, as in the original notebook, the `apply_func` is also used to check if changes have been set correctly.

fernandohds564 replied:
oh, right, only now I saw the apply_func just above. But is there a wait time between setting the values and printing the comparison on the apply_func function? because if there isn't, the user will always think it didn't apply and will call apply_func again, which is not optimal.

fernandohds564 reviewed (changes requested):
are all these user interactions really necessary? I tend to prefer having optional arguments for the script, instead of having to keep responding to questions along the running time. These questions make it harder to automatize things using the script... why did you decide to go this way? This is not the standard way we create scripts.

fernandohds564 commented:
Well, maybe the use will tell us.
Regarding the set-to-machine script, I think we can start this way and change things if we notice it is too cumbersome to run. Regarding the LOCO fitting script, I think I aggree with you @murilobalves. It is better to let it run freely, handling the options with the script's inputs.

fernandohds564 reviewed (commented):
the fitting script seems ok, but there still are two comments open in the apply-to-machine script.

vellosok75 commented:
> the fitting script seems ok, but there still are two comments open in the apply-to-machine script.

Yes. Couldn't work on them today but will do monday morning.

vellosok75 commented:
Guys, I changed the script and the base class methods for  allowing step-by-step setting of the families strengths in percentages. 

As you may remember, the main difficulty was that the set_delta_strength methods would always calculate the required change to apply based on the current state of the machine. I now added the option to use a fixed initial reference.

What do you think? @murilobalves @fernandohds564 @xresende

anacso17 commented:
Hi @vellosok75 , do you want to include this one in the next deploy? If yes, can you please update package version to `0.21.0`?

fernandohds564 commented:
@anacso17 , @vellosok75 in on vacation. I think we'll leave this one for a next opportunity.
